### PR TITLE
hypershift: promote to the OCP namespace

### PR DESCRIPTION
**Pull request by ironcladlou**
To get HyperShift into the OCP payload, we need to enable automatic
branch management by promoting images into the `ocp` namespace, which
this commit does.

This commit also stops promoting the test binary image because it's
only used internally by the jobs and not consumed by anybody externally.

### CODE REVIEW

File: core-services/image-mirroring/openshift/mapping_origin_4_10 [hunk 1]
```diff
@@ -133,6 +133,7 @@ registry.ci.openshift.org/origin/4.10:hyperconverged-cluster-operator quay.io/op
 registry.ci.openshift.org/origin/4.10:hyperconverged-cluster-operator-tests quay.io/openshift/origin-hyperconverged-cluster-operator-tests:4.10 quay.io/openshift/origin-hyperconverged-cluster-operator-tests:4.10.0 quay.io/openshift/origin-hyperconverged-cluster-operator-tests:latest
 registry.ci.openshift.org/origin/4.10:hyperconverged-cluster-webhook quay.io/openshift/origin-hyperconverged-cluster-webhook:4.10 quay.io/openshift/origin-hyperconverged-cluster-webhook:4.10.0 quay.io/openshift/origin-hyperconverged-cluster-webhook:latest
 registry.ci.openshift.org/origin/4.10:hyperkube quay.io/openshift/origin-hyperkube:4.10 quay.io/openshift/origin-hyperkube:4.10.0 quay.io/openshift/origin-hyperkube:latest
+registry.ci.openshift.org/origin/4.10:hypershift quay.io/openshift/origin-hypershift:4.10 quay.io/openshift/origin-hypershift:4.10.0 quay.io/openshift/origin-hypershift:latest
```
Comment (alvaroaleman) [inline core-services/image-mirroring/openshift/mapping_origin_4_10:136]: Why mirror to `quay.io/openshift/origin-hypershift`, shouldn't that be `registry.ci.openshift.org/hypershift/hypershift:latest`?

Comment (ironcladlou) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:136]: I'm not sure, that change was automatically generated by `make openshift-image-mirror-mappings` after I added the hypershift file...

File: core-services/image-mirroring/hypershift/mapping_hypershift_4_10 [hunk 2]
```diff
@@ -0,0 +1 @@
+registry.ci.openshift.org/ocp/4.10:hypershift registry.ci.openshift.org/hypershift/hypershift:4.10 registry.ci.openshift.org/hypershift/hypershift:latest
```
Comment (alvaroaleman) [inline core-services/image-mirroring/hypershift/mapping_hypershift_4_10:1]: This also needs a corresponding job, ref https://docs.ci.openshift.org/docs/how-tos/mirroring-to-quay/#configuring-mirroring-for-new-repository

Comment (ironcladlou) [reply core-services/image-mirroring/hypershift/mapping_hypershift_4_10]: Done

**Review by alvaroaleman — APPROVED** [review]

File: core-services/image-mirroring/openshift/mapping_origin_4_10 (hunk 1 above)
Comment (smarterclayton) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:136]: Everything that is part of origin/ocp is part of the same namespace and published the same way.  You can publish to multiple locations.  We don't allow images in OCP to have a canonical source that doesn't follow the pattern (it's a bit like RPMs - you can put an RPM in multiple repos, but the canonical location is decided by the publisher of the RPM).

**Pull request merged**